pkg/cloudprovider/providers/gce/gce_addresses.go

```diff
@@ -87,21 +87,19 @@ func (gce *GCECloud) GetRegionAddress(name, region string) (*compute.Address, er
 }
 
 // GetRegionAddressByIP returns the regional address matching the given IP
-// address. Note that if such an address does not exist, the function simply
-// returns a nil object. It does not return http.StatusNotFound like other
-// calls due to how it is implemented.
+// address.
 func (gce *GCECloud) GetRegionAddressByIP(region, ipAddress string) (*compute.Address, error) {
 	mc := newAddressMetricContext("list", region)
 	addrs, err := gce.service.Addresses.List(gce.projectID, region).Filter("address eq " + ipAddress).Do()
-	var match *compute.Address
-	if err == nil {
+	if err != nil {
+		return nil, mc.Observe(err)
+	}
+
+	for _, addr := range addrs.Items {
 		// There can at most be one match. No need to use the page token.
-		for _, addr := range addrs.Items {
-			if addr.Address == ipAddress {
-				match = addr
-				break
-			}
+		if addr.Address == ipAddress {
+			return addr, mc.Observe(err)
 		}
 	}
-	return match, mc.Observe(err)
+	return nil, mc.Observe(makeGoogleAPINotFoundError())
```

Comment:
IMO, the metrics should only reflect the GCP calls, so I'd suggest calling `mc.Observe(err)` between lines 93/94 instead of recording this made-up error.